packages/react-form-state/README.md

```diff
@@ -0,0 +1,947 @@
+# `@shopify/react-form-state`
+
+[![Build Status](https://travis-ci.org/Shopify/quilt.svg?branch=master)](https://travis-ci.org/Shopify/quilt)
+
+[![License: MIT](https://img.shields.io/badge/License-MIT-green.svg)](LICENSE.md) [![npm version](https://badge.fury.io/js/%40shopify%2Freact-formwell.svg)](https://badge.fury.io/js/%40shopify%2Freact-formwell.svg) [![npm bundle size (minified + gzip)](https://img.shields.io/bundlephobia/minzip/@shopify/react-formwell.svg)](https://img.shields.io/bundlephobia/minzip/@shopify/react-formwell.svg)
+
+Manage react forms tersely and type-safe with no magic.
+
+## Another form library?
+
+Good forms in React are surprisingly tough. A good api needs to support complex usecases with large numbers of fields split up across multiple components and explicit validation. It also needs to be quick to get started with and work for the simplest of forms cleanly. In meeting these goals, a package needs to balance developer experience, flexibility, and performance.
+
+Several popular community packages sit at different places on the spectrum for these requirements, and all are solid choices for many projects.
+
+**Examples**
+
+- [formik](https://github.com/jaredpalmer/formik) offers a terse but relatively unmagical way of managing form state with minimal dependencies.
+- [redux-form](https://redux-form.com/7.4.2/) offers a redux integrated solution that can be quite magical and give you the power of the redux devtools for debugging broken states.
+
+At Shopify we wanted our form solution to be true to the Web Foundations team's core values. We wanted it to be **explicit**, **type-safe**, and to **prioritize user experience** while **keeping code quality and developer happiness high**.
+
+As such the main difference in our solution is the total lack of magic. Form fields are given explicit handlers generated individually for each field and made available through a [render prop](https://reactjs.org/docs/render-props.html). To reduce boilerplate we've kept the generated `field` objects as easy to use as possible. Usually you can just splat them onto your inputs. We'll go into this in more detail in the [Usage](/#Usage) section.
```

Comment:
I like the word magic in the description, but I think it makes sense to use a more accurate term throughout the text. Maybe `implicit behaviour` or `automatic behaviour`?